### PR TITLE
Add EMA-GNN to Matbench Discovery leaderboard

### PR DESCRIPTION
**Pull request by submerged-in-matrix**
## Model description

EMA-GNN is a structural graph neural network that predicts formation energy per
atom directly from an unrelaxed crystal structure, with no relaxation at inference
time. It is a 6-seed ensemble of EMA-averaged models, 2.2M parameters each, trained
on Materials Project 2022 relaxed structures on a single consumer GPU.

Inference also uses test-time augmentation.

The architecture follows the structural GNN described in Merchant et al., Nature 624
(2023). That paper describes two models — the existing `gnome` entry on this
leaderboard is the other one, the NequIP-type interatomic potential submitted as
`IS2RE-SR` with targets `EF_G`. The structural GNN predicts energy directly from an
unrelaxed input and had not previously been submitted.

## Results

| Subset | F1 | DAF | Precision | Recall | MAE (eV/atom) | RMSE | R² |
|---|---|---|---|---|---|---|---|
| Full test set | 0.566 | 2.522 | 0.433 | 0.818 | 0.084 | 0.141 | 0.387 |
| Unique prototypes | 0.558 | 2.753 | 0.421 | 0.826 | 0.086 | 0.141 | 0.412 |

Scored with `matbench_discovery.metrics.stable_metrics` against the MP2020-corrected
reference columns, using hull-displacement
(`each_pred = each_true + e_form_pred - e_form_dft`) with the leaderboard's 5 eV/atom
outlier mask.  
Note ::: these are ingestion-computed values

## Why no ASE calculator

EMA-GNN cannot use `models/run_discovery.py`. That runner obtains a **total** energy
from an ASE calculator, applies `MaterialsProject2020Compatibility`, then subtracts
MP elemental references to derive `e_form_per_atom`
(`matbench_discovery/discovery.py:745`). EMA-GNN emits `e_form_per_atom` directly.
Routing it through the runner would require inverting the reference subtraction and
would apply MP2020 corrections on top of a quantity that already implicitly contains
them.

Predictions are therefore submitted as a hosted artifact, following the precedent of
the other archived direct-prediction `IS2E` entries. Happy to add a
`test_ema-gnn_discovery.py` if that would be preferred.

This PR also adds ema_gnn to ARCHIVED_DISCOVERY_MODELS in matbench_discovery/discovery.py. Without it, test_runnable_models_have_reproducible_runners fails for any active model with a pred_file that's in neither CALCULATORS nor the archived list. It can't be split into a prerequisite PR — test_model_registry_identity requires the archived key and the generated Model enum member to exist together, so each half fails without the other.

## Artifacts

Predictions and all six checkpoints are archived at
<https://doi.org/10.6084/m9.figshare.33111509>. There is no accompanying paper; the
Figshare item is the citable artifact. Code:
<https://github.com/submerged-in-matrix/gnome-repro-structural>

## Checklist

- [x] YAML metadata at `models/ema_gnn/ema-gnn.yml`
- [x] `models/ema_gnn/readme.md`
- [x] Predictions hosted with `size` and `md5` recorded
- [x] Checkpoints publicly available (CC-BY-4.0)
- [x] Code publicly available (MIT)
- [x] Schema validation passes
- [x] `enums.py` regenerated via `scripts/generate_model_enum.py` (not hand-edited)

### CODE REVIEW

File: site/src/figs/scatter-largest-fp-diff-each-error.jsonl [hunk 1]
```diff
@@ -11,6 +11,7 @@
...
...095,0.166,0.125,0.728,0.068,0.098,0.073,0.031,0.217,0.004,0.11,0.048,0.067,0.181,0.219,0.659,0.215,0.554,0.058,0.303,0.939,0.063,0.142,0.075,0.121,0.03,0.146,0.077,0.002,0.198,0.208,0.135,0.277,0.032,0.172,0.017,0.086,0.009,0.033,0.132,0.077,0.081,0.174,0.116,0.101,0.006,0.165,0.001,0.68,0.55,0.064,0.088,0.216,0.595,0.204,0.125,0.241,0.211,0.18,0.015,0.068,0.181,0.168,0.27,0.011,0.095,0.147,0.203,0.526,0.15,0.342,0.876,0.126,0.205,0.021,0.098,0.098,0.224,0.124,0.892,0.351,0.126,0.042,0.415,0.261,0.19,0.134,0.319,0.541,2.109,0.338,0.078,0.101,0.227,0.009,0.26,0.053,0.013,0.492,0.024,0.109,0.225,0.089,0.259,0.166,0.12,0.241,0.055,0.241,0.052,0.717,0.017,0.111,0.012,0.119,0.041,0.835,0.025,0.034,0.028,0.109,0.145,0.035,1.16,0.459,0.04,0.102,0.034,0.121,0.002,0.102,0.023,0.12,0.003,0.667,0.336,0.034,0.075,0.282,0.407,0.016,0.541,0.13,0.684,0.174,0.385,0.065,0.018,0.064,0.056,0.032,0.516,0.12,0.052,0.074,0.033,0.264,0.245,0.024,0.529,0.153,0.095,0.422,0.082,0.077,0.034,0.185,0.112,0.078,0.043,0.132,0.063,0.014,0.252,0.026,0.105,0.019,0.115,0.049,0.305,0.117,0.219,0.79,0.04,0.08,0.003,0.0,0.227,0.04,0.787,0.088,0.08,0.211,0.011,0.017,1.135,0.162,0.337,0.011,0.137,0.057,0.13,0.083,0.106]}
```
[the remark this answers is not in the document]
Comment (submerged-in-matrix) [reply site/src/figs/scatter-largest-fp-diff-each-error.jsonl:14]: mae here is over the largest-FP-diff subset, not the full test set — the two aren't comparable. Generated by scripts/ingest_model.py ema-gnn --payloads-only, so it uses the same canonical evaluation output as the YAML

File: site/src/figs/cumulative-precision-recall.jsonl [hunk 2]
```diff
@@ -11,6 +11,7 @@
...
...0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,0.9967,0.9882,0.98681,0.98585,0.98523,0.98113,0.97808,0.97285,0.96765,0.96024,0.94935,0.93834,0.92593,0.91147,0.90021,0.88923,0.87624,0.86867,0.85664,0.84198,0.83374,0.82275,0.8059,0.79251,0.78037,0.77153,0.76436,0.75401,0.74322,0.7349,0.72606,0.71656,0.70718,0.69775,0.6875,0.67629,0.6654,0.65243,0.63704,0.62267,0.60696,0.58892,0.57136,0.55069,0.52877,0.50414,0.47782,0.45041,0.42074,0.42075],"recall":[3e-05,6e-05,9e-05,0.00012,0.00015,0.00018,0.00021,0.00024,0.00027,0.00033,0.00036,0.00039,0.00045,0.00048,0.00054,0.00063,0.00069,0.00078,0.00087,0.00096,0.00108,0.0012,0.00135,0.0015,0.00168,0.00189,0.00213,0.00237,0.00264,0.00297,0.0033,0.00372,0.00413,0.00464,0.00518,0.00578,0.00647,0.00725,0.00812,0.00905,0.01004,0.01121,0.01252,0.01399,0.01558,0.01738,0.01933,0.02151,0.02388,0.0264,0.02918,0.03221,0.03548,0.03919,0.0433,0.04773,0.05292,0.05837,0.06418,0.07107,0.07844,0.08597,0.09453,0.10412,0.11515,0.12761,0.1408,0.15524,0.17169,0.18973,0.20944,0.2312,0.25517,0.28121,0.30943,0.34053,0.37346,0.40789,0.44595,0.48622,0.52769,0.57263,0.61734,0.66303,0.70708,0.7496,0.79035,0.82579,0.82582],"end":[65505,0.42075,0.82582]}
```
[the remark this answers is not in the document]
Comment (submerged-in-matrix) [reply site/src/figs/cumulative-precision-recall.jsonl:14]: The payload values are correct — 0.558 is the unique-prototypes F1, 0.566 is full-test. The PR body was quoting pre-ingestion figures and has been updated to match.

File: models/ema_gnn/ema-gnn.yml [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+      Recall: 0.826 # fraction
+      Accuracy: 0.797 # fraction
+      TP: 27563.0 # count
+      FP: 37921.0 # count
+      TN: 144193.0 # count
+      FN: 5811.0 # count
+      MAE: 0.086 # eV/atom
+      RMSE: 0.141 # eV/atom
+      R2: 0.412 # dimensionless
+      missing_preds: 12 # count
```
[the remark this answers is not in the document]
Comment (submerged-in-matrix) [reply models/ema_gnn/ema-gnn.yml:190]: The PR body was quoting pre-ingestion figures and has been updated to match.

**Review by janosh — APPROVED** [review]
Thanks for the contribution @submerged-in-matrix! 👍  valuable reproduction of the structural discovery model described in [Scaling deep learning for materials discovery](https://doi.org/10.1038/s41586-023-06735-9) (Nov 2023), provides useful historical context on its performance for comparing to other methods of its time. I verified the predictions and metrics, uploaded the artifacts to our Figshare, refreshed the figure payloads

**Comment by submerged-in-matrix** [comment]
Thank you for the review and for hosting the artifacts, @janosh! Happy to contribute — looking forward to seeing how the structural GNN baseline compares as the leaderboard evolves.
